Purchases/Public/RCPurchases.m

```diff
@@ -219,32 +209,53 @@ - (void)cachePurchaserInfo:(RCPurchaserInfo *)info {
     }
 }
 
-- (void)updateCaches
+- (void)updateCaches {
+    [self updateCachesWithCompletionBlock:nil];
+}
+
+- (void)updateCachesWithCompletionBlock:(RCReceivePurchaserInfoBlock _Nullable)completion
 {
     NSTimeInterval timeSinceLastCheck = -[self.cachesLastUpdated timeIntervalSinceNow];
     if (self.cachesLastUpdated != nil && timeSinceLastCheck < 60.) {
-        [self readPurchaserInfoFromCache];
```

Comment:
It worries me that the block will never complete in this case

Reply:
Hmm yea. I see a few other issues with the way I pushed debouncing into `RCBackend`. I need to think out all these methods to make sure:

1. We always call the completion block.
2. We don't spawn a bunch of backend requests. 
3. We don't refresh the cache too often.